lib/hyacinth/adapters/external_identifier_adapter/datacite.rb

```diff
@@ -17,96 +17,81 @@ def handles?(id)
           id =~ /^10\.[A-Za-z0-9]+\/[A-Za-z0-9\-]+/
         end
 
-        # Generates a new persistent id, ensuring that nothing currently uses that identifier.
+        # metadata is not required when minting a draft DOI.
+        # metadata is required when minting a findable DOI. This metadata must include all the
+        # properties required by DataCite.
+        # It is also possible to specify a (non-existent) DOI when minting. However, this is not
+        # currently supported in the code.
+        # @param digital_object [DigitalObject]
+        # @param target_url [String]
+        # @param doi_state [Symbol] doi_state can be set to one of the following: :draft, :findable, :registered
         # @return [String] a new id
-        def mint
-          # metadata is blank when minting
-          # TODO: allow existing object data to be pushed on mint
-          metadata = {}
-          # target is blank when minting
-          # TODO: allow existing object data to be pushed on mint
-          target_url = nil
-          # status is reserved when minting
-          # TODO: allow provided target
-          identifier_status = IDENTIFIER_STATUS[:reserved]
-          # mint_identifier returns a Hyacinth::Ezid::Doi
-          ezid_doi_instance = api_session.mint_identifier(DATACITE[:shoulder][:doi], identifier_status, target_url, metadata)
-          # if above method returned nil, call to EZID API was unsuccessful. Return nil to indicate failure
-          # if unsuccessful because of timeout, log faiilure and return nil
-          if api_session.timed_out?
-            log(:error, "#mint_and_store_doi: EZID API call to mint_identifier timed-out.")
-            return nil
+        def mint(digital_object: nil, target_url: nil, doi_state: :draft, **_args)
+          if doi_state != :draft && digital_object.nil?
+            Rails.logger.error "Hyacinth metadata required to mint DOI in #{doi_state} state"
+            return
           end
-          # if got a response from EZID server indicating failure, log info and return nil
-          if ezid_doi_instance.nil?
-            response = api_session.last_response_from_server
-            log(:error, "#mint_and_store_doi: EZID API call to mint_identifier was unsuccessful.")
-            log(:error, "#mint_and_store_doi: Response from EZID server follows:\n#{http_error_message(response)}")
-            return nil
+          datacite_data = Datacite::RestApi::V2::Data.new(@prefix)
+          datacite_data.update_properties(as_datacite_properties(digital_object, target_url)) if digital_object
```

Comment:
```suggestion
          datacite_data.update_properties(
            HyacinthMetadata.new(digital_object).as_datacite_properties(target_url)
          ) if digital_object
```
Optional suggestion, but if you do this then you can get rid of two levels of wrapper methods that do basically the same thing:

```
datacite.rb:
def as_datacite_properties(digital_object, target_url)
  HyacinthMetadata.as_datacite_properties(digital_object, target_url)
end
```

```
hyacinth_metadata.rb:
def self.as_datacite_properties(digital_object, target_url)
  HyacinthMetadata.new(digital_object).as_datacite_properties(target_url)
end
```

Might make the code easier to follow.

Reply:
In general it's a preferable encapsulation design to minimize constructor calls - they require a lot of mutual knowledge between classes - and a testing antipattern to call constructor-type methods in-line as arguments.  The better long-term change here, I think, would be to entirely replace the object-methods part of Hyacinth metadata with class/module functions, but this is a stepwise refactor of an existing implementation.